aspnet/mvc/views/razor.rst

```diff
@@ -1,8 +1,785 @@
+
+.. review: I copy/pasted from what was generated so my call differs from your version.
+
+.. code-block:: c#
+
+  public class _Views_Something_cshtml : RazorPage<dynamic>
+  {
+      public override async Task ExecuteAsync()
+      {
+          var output = "Hello World";
+  
+          WriteLiteral("/r/n<div>Output: ");
+          Write(output);
+          WriteLiteral("</div>");
+      }
+  }
+   
+:ref:`Razor-CustomCompilationService-label` explains how to view this generated class. Understanding how Razor generates code for a view will make it easier to understand how directives work.
+   
+``@using``
+^^^^^^^^^^^^^
+
+The ``@using`` directive will add the c# ``using`` directive to the generated razor page:
+
+.. review: You had @using System.Collections.Generic - but that's included in the Razor page.
+
+.. literalinclude:: razor/sample/Views/Home/Contact9.cshtml
+  :language: html
+  
+``@model``
+^^^^^^^^^^^^
+
+The ``@model`` directive allows you to specify the type of the model past to your Razor page. It uses the following syntax:
+
+.. code-block:: none
+
+  @model TypeNameOfModel<TModel> 
+
+For example, if you create a new ASP.NET Core MVC app with individual user accounts, the *Views/Account/Login.cshtml* Razor view file contains the follow model declaration:
+
+.. code-block:: c#
+
+  @model LoginViewModel
+
+In the class example in :ref:`Razor-Directives-label`, the class generated inherits from ``RazorPage<dynamic>``. By adding an ``@model`` you control what’s inherited. For example
+
+.. code-block:: c#
+
+  @model LoginViewModel
+  
+Generates the following class
+
+.. code-block:: c#
+
+ public class _Views_Account_Login_cshtml : RazorPage<LoginViewModel> 
+ 
+This allows you to access the strongly typed model in your Razor page through the ``Model`` property:
+
+.. code-block:: none
+
+  <div>The Login Email: @Model.Email</div>
+  
+Obviously you must pass the model from your controller to the view. See :ref:`Strongly-typed-models-keyword-label` for more information.
+
+``@inherits``
+^^^^^^^^^^^^^^^
+
+The ``@inherits`` directive is similar to the model directive. ``@inherits`` gives you full control of the class your Razor page inherits:
+
+.. code-block:: none
+
+ @inherits TypeNameOfClassToInheritFrom 
+ 
+.. review note: couldn't get non-generic to work. Test with http://localhost:24694/home/contact/11
+  public abstract class CustomRazorPage2 : RazorPage
+  {
+         public string CustomText { get; } = "CustomRazorPage2.";
+  }
+
+For instance, let’s say we had the following custom Razor page type:
+
+.. literalinclude:: razor/sample/Classes/CustomRazorPage.cs
+  :language: c#
+
+The following Razor would generate ``<div>Custom text: Hello World</div>``.
+
+.. literalinclude:: razor/sample/Views/Home/Contact10.cshtml
+  :language: html
+
+The ``@inherits`` keyword is not allowed when ``@model`` is used. You can pass a model as shown with the following Razor page (when passed "Rick@Example.com" in the model):
```

Comment:
- ...keyword is not allowed on the same page as an `@model` statement.
- Should use contoso instead of Example